### PR TITLE
Parallelize module download

### PR DESCRIPTION
**Pull request by nimrodkor**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Took a detour - instead of modifying the existing modules, I download them beforehand and leverage a cache to have the results accessible

### CODE REVIEW

**Comment by schosterbarak** [comment]
we should add/modify https://github.com/bridgecrewio/checkov/pull/1923/ performance test around this

**Comment by schosterbarak** [comment]
@nimrodkor  how the performance discussed at  https://github.com/bridgecrewio/checkov/issues/1531 after this PR?

File: tests/terraform/module_loading/test_registry.py [hunk 1]
```diff
@@ -26,14 +26,14 @@ def test_load_terraform_registry(self):
...
-                DEFAULT_EXTERNAL_MODULES_DIR,
-                "github.com/terraform-aws-modules/terraform-aws-security-group",
-            )
-            self.assertRegex(content.path(), f"^{expected_content_path}/v3.*")
+        content = registry.load(current_dir=self.current_dir, source=source, source_version="~> 3.0")
+        assert content.loaded()
+        expected_content_path = os.path.join(
+            self.current_dir,
+            DEFAULT_EXTERNAL_MODULES_DIR,
+            "github.com/terraform-aws-modules/terraform-aws-security-group",
```
Comment (schosterbarak) [inline tests/terraform/module_loading/test_registry.py:34]: I suggest to use a fork it so it will be long-lived and kept with the same code.

**Comment by nimrodkor** [comment]
> @nimrodkor how the performance discussed at #1531 after this PR?

See my comment on the issue 🤩

**Comment by schosterbarak** [comment]
> > @nimrodkor how the performance discussed at #1531 after this PR?
> 
> See my comment on the issue 🤩

rock start

**Review by gruebel — APPROVED** [review]
🚀

**Review by schosterbarak — APPROVED** [review]

**Pull request merged**